prime-router/src/main/kotlin/tokens/Server2ServerAuthentication.kt

```diff
@@ -22,7 +20,7 @@ import java.util.UUID
  * by a trusted party at the sender.  Per this guide:
  *    https://hl7.org/fhir/uv/bulkdata/authorization/index.html
  */
-class Server2ServerAuthentication(val metadata: Metadata) : Logging {
+class Server2ServerAuthentication(val workflowEngine: WorkflowEngine) : Logging {
```

Comment:
Why the change here? Also I am kind of suspicious of the original change as well...

Reply:
As discussed on video chat, this change avoids having to rebuild the workflow engine when it's already been created